src/transaction/log_prior_recv.cpp

```diff
@@ -58,11 +59,44 @@ namespace cublog
     std::unique_lock<std::mutex> ulock (m_messages_mutex);
     m_shutdown = true;
     ulock.unlock ();
-    m_messages_condvar.notify_one ();
+    m_messages_push_cv.notify_one ();
 
     m_thread.join ();
   }
 
+  void
+  prior_recver::wait_until_empty_and_pause ()
```

Comment:
I assume that when this function is called, no log records are added from the previous ATS. It's called between the new ATS connection and the previous one. If it's not met, we'll see it happens when appending log records because log records from two are not consecutive.